lib/index.js

```diff
@@ -85,6 +101,12 @@ function Socket (domain, type) {
     this.binding = nn.Socket(this.af_domain, this.protocol);
     this.queue = [];
 
+    opts = opts || {}
```

Comment:
If this was moved above line 28, then we wouldn't need `(opts || {})` on line 29.